codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -96,6 +104,71 @@ public DefaultHttp2Connection(boolean server) {
         streamMap.put(connectionStream.id(), connectionStream);
     }
 
+    /**
+     * Determine if {@link #close(Promise)} has been called and no more streams are allowed to be created.
+     */
+    final boolean isClosed() {
+        return closePromise != null;
+    }
+
+    @Override
+    public Future<Void> close(final Promise<Void> promise) {
```

Comment:
I'm just concerned about the complexity introduced by not using `ChannelPromise`.  It a real setup, it will always be a `ChannelPromise` so it would seem that some of this code could go away. How much harder does this make your tests? I seem to recall mocking `ChannelPromise` without too much of an issue, no?